neqo-transport/src/connection/mod.rs

```diff
@@ -1987,31 +1988,28 @@ impl Connection {
                 &self.address_validation,
                 version,
                 grease_quic_bit,
-            )?;
+            );
             let pn = Self::add_packet_number(
                 &mut builder,
                 tx,
                 self.loss_recovery.largest_acknowledged_pn(*space),
             );
-            let payload_start = builder.len();
+            if builder.remaining() < 2 {
```

Comment:
Please add a comment here as well that build_packet_header and add_packet_number will set the builder limit to  0 if there is not enough space for a packet header.